src/main/java/com/altenheim/kalender/controller/logicController/DateSuggestionController.java

```diff
@@ -0,0 +1,40 @@
+package com.altenheim.kalender.controller.logicController;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import com.altenheim.kalender.interfaces.IDateSuggestionController;
+import com.calendarfx.model.Entry;
+
+public class DateSuggestionController implements IDateSuggestionController 
+{
+    public Entry<String> getDateSuggestionFromEntryList(ArrayList<Entry<?>> input, LocalDateTime startSearchDateTime, int dateLenght) 
+    {
+        if (input == null || input.isEmpty() || input.get(input.size()-1).getEndAsLocalDateTime().isBefore(startSearchDateTime.plusMinutes((long) dateLenght + 1))) 
+            return null;
+
+        var runNumber = 0;
+        while (true) 
+        {
+            if (runNumber > input.size()-1)
+                return null;
+            
```

Comment:
statt den while mit true (kann man machen wenns nicht anders geht, aber besser ist schon ne echte bedingung) kann doch hier die abfrage des nachfolgenden ifs einfach als bedingung ein, oder? also while (runnumber > ...). Bzw. auf den ersten Blick sehe ich auch grad nicht was gegen ne ganz normale for Schleife spricht, ohne das oben die Variable initialisiert wird und im Loop inkrementiert wird.

Reply:
wenn du die laufvariable auf 1 setzt und dann nur nach inputsize abfragst (ohne -1) und bei den indexierungen von der L.v. einen abziehst (also andersrum) brauchst du die oben erwähnte abfrage empty nicht.